pandas/tseries/resample.py

```diff
@@ -854,9 +854,11 @@ def _get_time_bins(self, ax):
                                        closed=self.closed,
                                        base=self.base)
         tz = ax.tz
+        # do not call replace() because that will swallow the nanosecond part
```

Comment:
maybe a better comment as once we delete ``.replace`` its not clear what you are doing.